### PR TITLE
Prepare 2.0.1 release

### PR DESCRIPTION
**Pull request by rly**
## Motivation

If possible, let's merge these in first since they are ready to merge (or very close to ready). They are not urgent though, so if they are not merged by end of day, we can push them to the next release.

- [x] #1426 
- [x] #1417 
- [x] #1427
- [x] #1399 
- [x] #1402

## How to test the behavior?
```
Show how to reproduce the new behavior (can be a bug fix or a new feature)
```

## Checklist

- [] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

**Review by oruebel — DISMISSED** [review]

**Comment by oruebel** [comment]
#1426 I think is fine to merge. I think there are improvements that can be made, but I believe those are all internal changes that won't affect the user API so merging this now and enhancing it later seems fine.

#1417 is overall fine. I added a few comments, but I don't think those should take more than 1h to address. @rly  just ping me if you want me to take a crack at making the edits.

#1427 just ping me when I should review 

#1399 I updated the branch and completed the merge 

#1402 This PR is good to go. I updated it with dev. Just needs to be approved and merged.

**Comment by caniko** [comment]
I currently have a problem adding `pynwb` to my dependencies:
``` shell
poetry add --optional pynwb
Using version ^2.5.1 for pynwb

Updating dependencies
Resolving dependencies... (0.0s)

  SolverProblemError

  Because pynwb (2.5.1) depends on numpy (>=1.16,<1.21)
   and no versions of pynwb match >2.5.1,<3.0.0, pynwb (>=2.5.1,<3.0.0) requires numpy (>=1.16,<1.21).
  So, because spikeinterface depends on both numpy (^1.21) and pynwb (^2.5.1), version solving failed.
```
Would this PR solve the issue? Assuming this PR will lead to the new release being uploaded to PyPi.

**Comment by rly** [comment]
@caniko please install version 2.0.0. version 2.5.1 was an accidental release on pypi and has been yanked.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,10 +19,12 @@
   @weiglszonja (#1403)
 - Improved constructor docstrings for Image types. @weiglszonja (#1418)
 - Added documentation for exporting NWB files. @rly (#1417)
+- Improved documentation formatting. @bendichter (#1438)
+- Minor text fixes. @bendichter (#1437, #1400)
```
Comment (oruebel) [inline CHANGELOG.md:24]: I think the Changelog may be missing the updates to the tutorial that were merged today

**Review by oruebel — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (rly) [reply CHANGELOG.md:24]: The tutorial updates are included in the list items above this. The two items added here were missing.

**Pull request merged**